File: js/axios/webhooks/fetchWebhooks.js
import axios from "axios";
import "dotenv/config";

const accessKey = process.env.ACCESS_TOKEN;
const apiUrl = process.env.production
  ? "https://api.theauthapi.com"
  : process.env.TESTING_URL;

async function fetchWebhooks(projectId) {
  try {
    const { data } = await axios
      .get(`${apiUrl}/webhooks?projectId=${projectId}`, {
        headers: {
          ContentType: "application/json",
          "x-api-key": accessKey,
        },
      });
    return data;
  } catch (error) {
    if (error.response) {
      console.log(error.response.data);
    } else {
      // handle other errors
    }
  }
}

(async () => {
  const webhooks = await fetchWebhooks(process.env.PROJECT_ID);
  console.log(webhooks);
})();

File: python/webhooks/fetch_webhooks.py
import requests
from requests.exceptions import HTTPError

api_url = "https://api.theauthapi.com/webhooks"
access_key = "REPLACE_WITH_YOUR_ACCESS_KEY"

headers = {
    "x-api-key": access_key,
}

def fetch_webhooks(project_id):
    try:
        response = requests.get(api_url, headers=headers, params={ 'projectId': project_id })
        response.raise_for_status()
        return response.json()
    except HTTPError as error: 
        # handle error
        print(error)


project_id = 'REPLACE_WITH_YOUR_PROJECT_ID'
webhooks = fetch_webhooks(project_id)
print(webhooks)
